operators/src/processing/point_in_polygon.rs

```diff
@@ -115,28 +119,69 @@ impl PointInPolygonFilterProcessor {
         Self { points, polygons }
     }
 
-    fn filter_points(
-        points: &MultiPointCollection,
-        polygons: &MultiPolygonCollection,
-        initial_filter: &BooleanArray,
-    ) -> Result<BooleanArray> {
-        let mut filter = Vec::with_capacity(points.len());
+    fn filter_parallel(
+        tester: &Arc<PointInPolygonTester>,
+        points: &Arc<MultiPointCollection>,
+        thread_pool: &ThreadPoolContext,
+    ) -> Vec<bool> {
+        let parallelism = thread_pool.degree_of_parallelism();
+        let chunk_size = (points.len() as f64 / parallelism as f64).ceil() as usize;
+
+        let mut result = vec![false; points.len()];
+
+        thread_pool.scope(|scope| {
+            let num_features = points.len();
+            let feature_offsets = points.feature_offsets();
+            let time_intervals = points.time_intervals();
+            let coordinates = points.coordinates();
+
+            for (chunk_index, chunk_result) in (&mut result).chunks_mut(chunk_size).enumerate() {
+                let feature_index_start = chunk_index * chunk_size;
+                let features_index_end = min(feature_index_start + chunk_size, num_features);
+                let tester = tester.clone();
+
+                scope.compute(move || {
+                    for (
+                        feature_index,
+                        ((coordinates_start_index, coordinates_end_index), time_interval),
+                    ) in two_tuple_windows(
+                        feature_offsets[feature_index_start..=features_index_end]
+                            .iter()
+                            .map(|&c| c as usize),
+                    )
+                    .zip(time_intervals[feature_index_start..features_index_end].iter())
+                    .enumerate()
+                    {
+                        let is_multi_point_in_polygon_collection = coordinates
+                            [coordinates_start_index..coordinates_end_index]
+                            .iter()
+                            .any(|coordinate| {
+                                tester.is_coordinate_in_any_polygon(coordinate, time_interval)
+                            });
 
-        let tester = PointInPolygonTester::new(polygons);
+                        chunk_result[feature_index] = is_multi_point_in_polygon_collection;
+                    }
+                });
```

Comment:
maybe we should add a TODO to improve further by looking up coordinates independent of their feature in threads

Reply:
b4087ec0a91cee8d5f4773e37be84c6839cd51cc